common/ledger/util/leveldbhelper/leveldb_provider.go

```diff
@@ -149,6 +166,32 @@ func (h *DBHandle) Delete(key []byte, sync bool) error {
 	return h.db.Delete(constructLevelKey(h.dbName, key), sync)
 }
 
+// DeleteAll deletes all the keys that belong to the channel (dbName).DeleteAll
+func (h *DBHandle) DeleteAll() error {
+	iter := h.GetIterator(nil, nil)
+	defer iter.Release()
+
+	numKeys := 0
+	batch := NewUpdateBatch()
+	for iter.Next() {
+		key := iter.Key()
+		batch.Delete(key)
+		numKeys++
+		if batch.Len() >= maxBatchSize {
+			if err := h.WriteBatch(batch, true); err != nil {
+				return err
+			}
+			batch = NewUpdateBatch()
+			sleepTime := time.Duration(batchesInterval)
+			logger.Infof("Sleep for %d milliseconds between batches of deletion. Entries have been removed for channel %s: %d", sleepTime, h.dbName, numKeys)
```

Comment:
The sleep is between batches so that the deletion does not throttle the other operations by causing a sudden burst of I/O. Sleep is added because Remove is a non-blocking method. Need to reconsider if Remove is a blocking method.